internal/certgen/certgen.go

```diff
@@ -326,3 +487,48 @@ func bigIntHash(n *big.Int) []byte {
 	h.Write(n.Bytes())
 	return h.Sum(nil)
 }
+
+func serviceNames(service, namespace string) []string {
+	return []string{
+		service,
+		fmt.Sprintf("%s.%s", service, namespace),
+		fmt.Sprintf("%s.%s.svc", service, namespace),
+		fmt.Sprintf("%s.%s.svc.cluster.local", service, namespace),
+	}
+}
+
+func writePEMFile(filename string, data []byte) error {
+	_, err := os.Stat(filename)
+	if err != nil {
+		// Can't stat the file, so we'll create it
+		err = ioutil.WriteFile(filename, data, 0644)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Created %s\n", filename)
+		return nil
+	}
+	return fmt.Errorf("can't overwrite %s", filename)
+
+}
+
+func writeSecret(filename string, secret *corev1.Secret) error {
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+	_, err := os.Stat(filename)
+	if err != nil {
```

Comment:
putting the success case _inside_ the conditional block isn't considered good form, its better to use a guard clause to bail out early if the precondition is not met.

In this case I'd recommend making a helper, say, `createFile(path, force) (f, err)` where createFile creates path iff it does not exist and force is false, otherwise returns with an appropriate error message, this will let you write something like this

f, err := createFile(filename, false) // never overwrite
if err != nil {
    return err
}